daemon/mgr/container.go

```diff
@@ -1175,6 +1179,57 @@ func (mgr *ContainerManager) Restart(ctx context.Context, name string, timeout i
 	return mgr.start(ctx, c, "")
 }
 
+// DisconnectContainerFromNetwork disconnects the given container from
+// given network
+func (mgr *ContainerManager) DisconnectContainerFromNetwork(ctx context.Context, containerName, networkName string, force bool) error {
+	c, err := mgr.container(containerName)
+	if err != nil {
+		// TODO(ziren): if force is true, force delete endpoint
+		return err
+	}
+
+	// Get network
+	network, err := mgr.NetworkMgr.Get(ctx, networkName)
+	if err != nil {
+		return fmt.Errorf("failed to get network %s when disconnecting container %s: %v", networkName, c.Name(), err)
+	}
+
+	networkSettings := c.meta.NetworkSettings
+	if networkSettings == nil || networkSettings.Networks == nil {
```

Comment:
I tested that there is no need to check if networkSettings.Networks equals to nil, if we checked `epConfig, ok := networkSettings.Networks[network.Name]`. Since we could get a value by the following way if it is nil:
```
value, exist := networkSettings.Networks[network.Name]
```
even if `networkSettings.Networks` is nil.